node/src/WebRtcTransport.ts

```diff
@@ -11,25 +11,36 @@ import {
 } from './Transport';
 import { WebRtcServer } from './WebRtcServer';
 import { SctpParameters, NumSctpStreams } from './SctpParameters';
+import { Either } from './utils';
 
-export type WebRtcTransportOptions =
+export type WebRtcTransportListenIndividual =
 {
-	/**
-	 * Instance of WebRtcServer. Mandatory unless listenIps is given.
-	 */
-	webRtcServer?: WebRtcServer;
 	/**
 	 * Listening IP address or addresses in order of preference (first one is the
 	 * preferred one). Mandatory unless webRtcServer is given.
 	 */
-	listenIps?: (TransportListenIp | string)[];
+	listenIps: (TransportListenIp | string)[];
 
 	/**
 	 * Fixed port to listen on instead of selecting automatically from Worker's port
 	 * range.
 	 */
 	port?: number;
+}
 
+export type WebRtcTransportListenServer =
+{
+	/**
+	 * Instance of WebRtcServer. Mandatory unless listenIps is given.
+	 */
+	webRtcServer: WebRtcServer;
+}
+
+export type WebRtcTransportListen =
+	Either<WebRtcTransportListenIndividual, WebRtcTransportListenServer>;
```

Comment:
OMG true. We don't need any new Either.

Reply:
We do, otherwise fields of both will be combined. We don't want that.
We want either fields from one and none from the other or the opposite.

Reply:
Don't wrap listenIps and webRtcServer into objects but just add a listen: listenIps | webRrcServer; and it should work, although we are changing the api if so...

So your Either doesn't imply any api change right? If so let's go

Reply:
Yes, Either ensures API stays the same. `listen: listenIps | webRtcServer` isn't a full solution because there is also a port, which doesn't make sense in case of `webRtcServer`.

Just apply these changes and see what compiles and what no longer does.